src/utils/get-babel-rc.ts

```diff
@@ -6,11 +6,21 @@ import {
   PACKAGE_JSON,
   BABEL_CONFIG_KEY,
 } from './constants';
+import { importJson5 } from './imports';
+
+// Don't import JSON5 parser unless specifically asked to parse a babel config file
+let parse;
+function parseBabelConfig(config) {
+  if (parse == null) {
+    parse = importJson5().parse;
+  }
+  return parse(config);
+}
 
 const babelReaders = [
   {
     basename: BABELRC_FILENAME,
-    read: f => JSON.parse(readFileSync(f, 'utf8')),
+    read: f => parseBabelConfig(readFileSync(f, 'utf8')),
```

Comment:
I'd actually do a helper maybe in `src/utils/parse-json.ts`, even tho for now we'll use this only for babelrc:

```js
import { importJson5 } from './imports'

let parse: any

export default (data: string) => {
    if (!parse) {
	    try {
	    	return JSON.parse(res);
        } catch(err) {}
        parse = importJson5().parse;
    }
    return parse(data)
}
```

Then change the error message in parseJson to be more generic.

Reply:
... so that it loads it ONLY if the default `JSON.parse` would fail

Reply:
I disagree with a parse function that does both; I think it's much better to be explicit which JSON spec is intended to be used.

For example, if this function is used for some other file which must be pure JSON (e.g. parsed by another module that enforces JSON-only), it would be better to fail here as well.

If you still want this change I'll add it, but wanted to give my thoughts first.

Reply:
I see your point, I guess here it's just a matter of name. It could be named `parseJson5` or so. Then we either use `JSON.parse()` when it has to be pure JSON, else parseJson5.

The main thing here is that users who have a babelrc with valid JSON shouldn't need to install `json5` as a dep...

@ahnpnl @GeeWee thoughts?

Reply:
I like the idea of @chandlerprall  to have separate parse functions and depending on which type of JSON the user passes in then the code calls the according parse function to deal with the file.